tesla_powerwall/api.py

```diff
@@ -15,17 +16,18 @@ def __init__(
         self,
         endpoint: str,
         timeout: int = 10,
-        http_session: Optional[requests.Session] = None,
+        http_session: Optional[aiohttp.ClientSession] = None,
         verify_ssl: bool = False,
         disable_insecure_warning: bool = True,
     ) -> None:
         if disable_insecure_warning:
             disable_warnings(InsecureRequestWarning)
 
         self._endpoint = self._parse_endpoint(endpoint)
-        self._timeout = timeout
-        self._http_session = http_session if http_session else requests.Session()
-        self._http_session.verify = verify_ssl
+        self._timeout = aiohttp.ClientTimeout(total=timeout)
+        self._owns_http_session = False if http_session else True
+        self._http_session = http_session if http_session else aiohttp.ClientSession()
```

Comment:
I think we need the unsafe cookie jar to allow the ip based cookies here as well
https://github.com/home-assistant/core/pull/107164/files#diff-f1eecf25dc296b0bce35cd4f3f74c6b88f3396694f986c97d804bf4a33a62b97R125